nextcloud/nextcloud_manager.py

```
@@ -37,58 +39,126 @@ def initialize_connection() -> None:
     )
 
 
+def _check_initialized(method):
+    """
+    Wrapper to check whether the Nextcloud connection was initialized.
+    :raises Exception: If connection has not been initialized
+    """
+
+    def wrapper(*args, **kwargs):
+        try:
+            nc
+        except NameError:
+            raise Exception(
+                "Call initialize_connection() first to initialize the Nextcloud connection"
+            )
+        return method(*args, **kwargs)
+
+    return wrapper
+
+
+@_check_initialized
 def upload_file(nc_path: str, local_path: PathLike[bytes] | str) -> None:
     """
-    Uploads a file to the Nextcloud server
+    Uploads a file to the Nextcloud server. Overwrites any existing file with same path+name in nextcloud. Directory that should contain the file must already exist.
```

This function should probably throw an exception instead of silently overwriting existing files. At least make it a parameter (i.e. `overwrite=True`)